diofant/polys/rootoftools.py

```diff
@@ -218,6 +218,11 @@ def _eval_conjugate(self):
             ci = self.index + 2*((self.index - nreals + 1) % 2) - 1
             return self._new(self.poly, ci)
 
+    def as_real_imag(self, deep=True, **hints):
+        if (cself := self._eval_conjugate()) is not None:
+            return ((self + cself)/2, (self - cself)/I/2)
+        return super().as_real_imag(deep=deep, **hints)
```

Comment:
@oscarbenjamin, this approach should be less costly for SymPy too, c.f. https://github.com/sympy/sympy/pull/26828 (but to work correctly, https://github.com/sympy/sympy/issues/14293 should be solved in SymPy, not just closed).

BTW, which application do you see for Poly.real_imag() in multivariate case?